### PR TITLE
[JS] Topic: Promises in Javascript

### LINKED ISSUE #1241: JS : Promises
# Promises
---
Promises in Javascript

# Directory
---
```
Topics/10.JSON
```
 
# Description 📜
---
- What is a Promise in Javascript?
- Why do we need Promises?
- How to create a Promise? (Syntax)
- Chained Promises
- Methods in Promises
- Examples

# Domain of Contribution 📊
---
- [x] JavaScript

# Format
---
- [x] Documentation

### PR DESCRIPTION
**Pull request by aniketpathak028**
## Description 📜

Created a tutorial for Promises in Javascript and elaborated upon the following points:

- What is a Promise in Javascript?
- Why do we need Promises?
- How to create a Promise? (Syntax)
- Chained Promises
- Methods in Promises
- Examples

Fixes  #1241

<hr>

## Type of change 📝

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] JavaScript

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Comment by aniketpathak028** [comment]
@paramsiddharth sorry for bothering you multiple times today 😅, 
I have completed this Promises PR please suggest changes if required!
Thanks 😇

**Comment by aniketpathak028** [comment]
@creatoon @dakshsethi pls review this PR

**Comment by paramsiddharth** [comment]
@Creatoon @dakshsethi @s4-2 Please review.

**Review by paramsiddharth — APPROVED** [review]

**Review by s4-2 — APPROVED** [review]
Looks good to me, well done:)

File: JavaScript/Topics/12.Promises/README.md [hunk 1]
```diff
@@ -0,0 +1,222 @@
+# Promises
+---
```
Comment (dakshsethi) [inline JavaScript/Topics/12.Promises/README.md:2]: Make it `# Promises in JavaScript`

File: JavaScript/Topics/12.Promises/README.md [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+  .then(handleResolvedC, handleRejectedC);
+```
+
+## Methods in Promises
+---
+
+### 1. Promise.all(iterable)
+---
+Wait for all promises to be resolved, or for any to be rejected. If the returned promise resolves, it is resolved with an aggregating array of the values from the resolved promises, in the same order as defined in the iterable of multiple promises. If it rejects, it is rejected with the reason from the first promise in the iterable that was rejected.
+
```
Comment (dakshsethi) [inline JavaScript/Topics/12.Promises/README.md:193]: Show code snippets of all  `Methods in Promises`

**Review by dakshsethi — CHANGES_REQUESTED** [review]

File: JavaScript/Topics/12.Promises/README.md [hunk 3]
```diff
@@ -0,0 +1,222 @@
+# Promises
+---
+The Promise object represents the eventual completion (or failure) of an asynchronous operation and its resulting value.
+
+Example:
+```js
+let myPromise = new Promise(function(myResolve, myReject) {
+// "Producing Code" (May take some time)
```
Comment (Creatoon) [inline JavaScript/Topics/12.Promises/README.md:8]:
```suggestion
// "Processing Code" (May take some time)
```

File: JavaScript/Topics/12.Promises/README.md [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+
+  myResolve(); // when successful
+  myReject();  // when error
+});
+
+// "Consuming Code" (Must wait for a fulfilled Promise)
+myPromise.then(
+  function(value) { /* code if successful */ },
```
Comment (Creatoon) [inline JavaScript/Topics/12.Promises/README.md:16]:
```suggestion
  function(value), { /* code if successful */ },
```

File: JavaScript/Topics/12.Promises/README.md [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+const y = "javascript"
+if(x === y) {
+	resolve();
+} else {
+	reject();
+}
+});
+
+promise.
+	then(function () {
```
Comment (Creatoon) [inline JavaScript/Topics/12.Promises/README.md:83]:
```suggestion
	then(function (result) {
```

File: JavaScript/Topics/12.Promises/README.md [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+		console.log('Success, You are a coder');
+	}).
+	catch(function () {
```
Comment (Creatoon) [inline JavaScript/Topics/12.Promises/README.md:86]:
```suggestion
	catch(function (error) {
```

**Review by Creatoon — CHANGES_REQUESTED** [review]

**Comment by aniketpathak028** [comment]
@Creatoon @dakshsethi made all those changes that were suggested! Please approve. 🙂
Thanks

**Review by Creatoon — APPROVED** [review]

**Pull request merged**